Library/Homebrew/sandbox.rb

```diff
@@ -91,24 +98,36 @@ def exec(*args)
     @start = Time.now
     safe_system SANDBOX_EXEC, "-f", seatbelt.path, *args
   rescue
-    if ARGV.verbose?
-      ohai "Sandbox profile:"
-      puts @profile.dump
-    end
+    @failed = true
     raise
   ensure
     seatbelt.unlink
-    unless @log.nil?
+    if @log
       sleep 0.1 # wait for a bit to let syslog catch up the latest events.
       syslog_args = %W[
-        -F '$((Time)(local))\ $(Sender)[$(PID)]:\ $Message'
+        -F $((Time)(local))\ $(Sender)[$(PID)]:\ $(Message)
         -k Time ge #{@start.to_i}
+        -k Message S deny
         -k Sender kernel
         -o
         -k Time ge #{@start.to_i}
+        -k Message S deny
         -k Sender sandboxd
       ]
-      quiet_system "syslog #{syslog_args * " "} | grep deny > #{@log}"
+      logs = Utils.popen_read("syslog", *syslog_args)
+      unless logs.empty?
+        @log.write Sandbox.log_warning_message
+        @log.write logs
+
+        if @failed && ARGV.verbose?
+          ohai "Sandbox log:"
+          puts Sandbox.log_warning_message
+          puts logs
+          ohai "Sandbox profile:"
+          puts @profile.dump
```

Comment:
Yeh, I guess I personally find this output kinda noise and hard to follow. Maybe we could add a few newlines or separators in here to split it up a bit if we leave it in?